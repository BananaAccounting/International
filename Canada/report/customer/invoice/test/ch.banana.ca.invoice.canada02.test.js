// Copyright [2018] [Banana.ch SA - Lugano Switzerland]
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// 
//     http://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//


// @id = ch.banana.ca.invoice.canada01.test
// @api = 1.0
// @pubdate = 2018-04-04
// @publisher = Banana.ch SA
// @description = <TEST ch.banana.ca.invoice.canada01.js>
// @task = app.command
// @doctype = *.*
// @docproperties = 
// @outputformat = none
// @inputdataform = none
// @includejs = ../ch.banana.ca.invoice.canada01.js
// @timeout = -1


/*

  SUMMARY
  -------
  Javascript test
  1. Open the .ac2 file
  2. Execute the .js script
  3. Save the report



  virtual void addTestBegin(const QString& key, const QString& comment = QString());
  virtual void addTestEnd();

  virtual void addSection(const QString& key);
  virtual void addSubSection(const QString& key);
  virtual void addSubSubSection(const QString& key);

  virtual void addComment(const QString& comment);
  virtual void addInfo(const QString& key, const QString& value1, const QString& value2 = QString(), const QString& value3 = QString());
  virtual void addFatalError(const QString& error);
  virtual void addKeyValue(const QString& key, const QString& value, const QString& comment = QString());
  virtual void addReport(const QString& key, QJSValue report, const QString& comment = QString());
  virtual void addTable(const QString& key, QJSValue table, QStringList colXmlNames = QStringList(), const QString& comment = QString());

**/

// Register test case to be executed
Test.registerTestCase(new ReportCustomerInvoiceCanada01());

// Here we define the class, the name of the class is not important
function ReportCustomerInvoiceCanada01() {

}

// This method will be called at the beginning of the test case
ReportCustomerInvoiceCanada01.prototype.initTestCase = function() {

}

// This method will be called at the end of the test case
ReportCustomerInvoiceCanada01.prototype.cleanupTestCase = function() {

}

// This method will be called before every test method is executed
ReportCustomerInvoiceCanada01.prototype.init = function() {

}

// This method will be called after every test method is executed
ReportCustomerInvoiceCanada01.prototype.cleanup = function() {

}

ReportCustomerInvoiceCanada01.prototype.testReport = function() {
   
  Test.logger.addComment("Test ch.banana.ca.invoice.canada01.js");

  var fileAC2 = "file:script/../test/testcases/invoices_universal.ac2";
  var banDoc = Banana.application.openDocument(fileAC2);
  
  //Banana.console.log("bandoc : " + banDoc);
  if (!banDoc) {return;}

  Test.logger.addSection("Invoice tests");


  //Invoice 36
  Test.logger.addSubSection("Test Invoice 36");
  aggiungiReport("36", "Test Invoice 36");

}

//Function that creates the report for the test
function aggiungiReport(invoiceNumber, reportName) {
  
  //Set params (normally are taken from settings)
  var param = {};
  param.print_header = true;
  param.print_paymentslip = false;
  param.font_family = '';
  param.color_1 = '';
  param.color_2 = '';
  param.color_3 = '';
  param.color_4 = '';
  param.color_5 = '';

  //Banana.console.log(JSON.stringify(jsonInvoice, "", ""));

  var jsonInvoice = getJsonInvoice(invoiceNumber);
  var invoiceReport = printInvoice(jsonInvoice, invoiceReport, param);

  Test.logger.addReport(reportName, invoiceReport);
}


function getJsonInvoice(invoiceNumber) {
  var file;
  var parsedfile;
  var jsonInvoice;
  

  if (invoiceNumber === "36") {
    file = Banana.IO.getLocalFile("file:script/../test/testcases/json_invoice_36.json");
    parsedfile = JSON.stringify(file.read(), "", "");
    jsonInvoice = JSON.parse(parsedfile);
    // Banana.console.log(jsonInvoice);
  }

  return jsonInvoice;
}
