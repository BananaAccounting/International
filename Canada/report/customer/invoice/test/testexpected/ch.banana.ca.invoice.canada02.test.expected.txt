%%info_test_name{D:/github/International/Canada/report/customer/invoice/test/ch.banana.ca.invoice.canada02.test.js}
%%info_test_app{Banana90}{9.0.2}
%%info_test_os{Windows 10 (10.0)}
%%info_test_qt{5.9.2}
%%info_test_date{2018-02-14}{08:49:22.129}
\documentclass{report}
\usepackage{color}
\usepackage{adjustbox}
\usepackage[margin=1cm]{geometry}
\begin{document}
\newenvironment{tablevalue}[2]
   {\textbf{Table: #1}\\\begin{adjustbox}{max width=\textwidth}\tabular{#2}}
   {\endtabular\end{adjustbox}}
\newenvironment{reportvalue}[1]{\textbf{Report: #1}\\}{\\}
\newcommand{\info}[2]{\textit{#1: #2}\\}
\newcommand{\fatalerror}[1]{\textbf{\textcolor[rgb]{1,0,0}{Fatal error: #1}}\\}
\newcommand{\keyvalue}[2]{\textbf{Keyvalue: #1} #2\\}
\newcommand{\excltest}[1]{\\Excluded from compare: #1}
\newenvironment{testcase}[1]{\section*{Test: #1}}{\newpage}
\newenvironment{test}[1]{\section*{Test case: #1}}{}
%Begin test data
\begin{testcase}{ReportCustomerInvoiceCanada01}
\begin{test}{test1}
\keyvalue{Result}{Passed}
%%info_test_time_elapsed{00:00:00.000}
\end{test}
\begin{test}{test2}
\keyvalue{Result}{Passed}
%%info_test_time_elapsed{00:00:00.000}
\end{test}
\begin{test}{testReport}
%Test ch.banana.ca.invoice.canada01.js
\section{Invoice tests}
\subsection{Test Invoice 36}
\begin{reportvalue}{Test Invoice 36}
\begin{tabular}{ l  l }
Banana.ch SA  & INVOICE \\
\end{tabular}
\begin{tabular}{ l  l  l }
Via Trevano 7a  \\
Lugano, Ticino 6900 Switzerland   \\
 & DATE: \\
INVOICE #: \\
PAYMENT: \\
 & 29.01.2018 \\
36 \\
Net 30 Days \\
\\
\end{tabular}
\begin{tabular}{ l  l  l }
\multicolumn{3}{l}{  \\
Bill To: \\
  \\
}\\
Marco Alberti \\
V. CampurÃ©ta 40 \\
Novazzano, Ticino 6883 \\
Switzerland \\
 & Phone: \\
Fax: \\
Email: \\
Website: \\
 & 416-417-6521 \\
416-417-8846 \\
info@banana.ch \\
www.robertbrennanaccountant.ca \\
\\
\end{tabular}
\begin{tabular}{}
\end{tabular}
\begin{tabular}{ l  l  l }
FOR THE FOLLOWING PROFESSIONAL SERVICES RENDERED  & \multicolumn{2}{l}{AMOUNT }\\
Preparation of unaudited financial statements and corporate income tax return for the year ended Oct 31, 2017 \\
\\
 & $  & 1'000.00 \\
 & \multicolumn{2}{l}{}\\
 & \multicolumn{2}{l}{}\\
 & \multicolumn{2}{l}{}\\
 & \multicolumn{2}{l}{}\\
 & \multicolumn{2}{l}{}\\
 & \multicolumn{2}{l}{}\\
SUBTOTAL  & $  & 1'000.00 \\
HST # 867136624RP0001  & $  & 130.00 \\
INVOICE TOTAL  & $  & 1'130.00 \\
LESS RECEIVED ON ACCOUNT  &  & \\
BALANCE DUE  & $  & 1'130.00 \\
\end{tabular}

\end{reportvalue}
\keyvalue{Result}{Passed}
%%info_test_time_elapsed{00:00:00.297}
\end{test}
%%info_test_time_elapsed{00:00:00.297}
\end{testcase}
%%info_test_time_elapsed{00:00:00.547}
\end{document}
